### ISSUE TITLE
output json format results for data loading and for data analysis for d3

### ISSUE THREAD

**Issue by didillysquat** [didillysquat/SymPortal_framework#66]
For creating the d3 html output plots it is easiest to have outputs in json format. This way we don't have to mess around with typing the outputs.

**Comment by didillysquat** [comment]
Implement with #62

**Comment by didillysquat** [comment]
Likely best if we put these into a seperate 'html' directory. We can also have the colour dictionary in here too. #65

**Comment by didillysquat** [comment]
still need to:

- write out the colour file from the main seq plotting code
- write out the colour file from the pre-med seq plotting code
- write out the pre-med seq abundances in the form of js methods to load json data

**Comment by didillysquat** [comment]
It would be good to have the d3 associated resource files for the pre-med seqs available in the html directory that will be produced when performing an analysis or a stand alone analysis output. As part of the normal seq and profile output the post-med seq resources for the d3/html will be provided but the pre-med seqs will not. As the pre-med sequencing data is not retained in the database (currently) we would have to rely on knowing or inferring the path of where the pre-med seq data was written out at the time of initial data_loading. This could get very messy and certainly unreliable given that the outputs may get moved around (i.e. when delivered to users from a remotely run analysis). I think therefore that it might be time to address an Elephant in the room and move towards attaching the pre-MED data to the DataSetSample objects. The best way to do this will likely be to use the ReferenceSequence object for this purpose, i.e. have every sequence found turned into a ReferenceSequence. We will have to revisit the code we use when creating RS objects to make sure that we do this in a robust manner, i.e. one that covers the occasion where one sequences is a subset of another, or vice versa. The incorporation of pre_med sequences to the database has the potential for being quite large in scope in terms of the operations it will effect. As such I will make it a separate project with its own issues associated.

To prevent the opening up of too many wounds in SymPortal at once, I think its likely a good idea to work through the current issues related to outputting the meta, and abund outputs seperately, as well as producing the d3 resources (without pre-med for the analyses). Get these complete, and then we can work on incorporating the pre-med info into the database, into the standard outputs and into the html outputs.

In summary, and specifically for this issue, let's ignore pre-med output info when outputting from a data analysis.
